### ISSUE TITLE
India - Optional Holiday and Other Holidays which depend on Lunar dates

### ISSUE THREAD

**Issue by rishabmehta7** [vacanza/holidays#413]
Hi Guys,

Is there an option to specify optional holidays based or religion?
To give you an example Christmas is a public holiday irrespective of the religion, but something like 'Mahaveer Jayanti' is a holiday for 'Jains' (People following Jainism). Similarly there are plenty of examples for Hindus, Sikhs, Muslims etc.

Also most of these very much depend on lunar dates and not on Georgian calendar. Is there a library that you are aware of that can provide the correct phase of moon? I can use those to specify the other major Holidays in India.

Rishab

**Comment by xames3** [comment]
I would like to work on and submit a patch for an Indian holiday for 19th Feb, which is Shivaji Jayanti in Maharashtra, India. Is it fine if I submit a patch as part of this PR?

**Comment by mborsetti** [comment]
Rashid,

> Is there an option to specify optional holidays based or religion? To give you an example Christmas is a public holiday irrespective of the religion, but something like 'Mahaveer Jayanti' is a holiday for 'Jains' (People following Jainism). Similarly there are plenty of examples for Hindus, Sikhs, Muslims etc.

That's an interesting use case and for the moment only states or provinces are allowed, but of course code could be modified.

> Also most of these very much depend on lunar dates and not on Georgian calendar. Is there a library that you are aware of that can provide the correct phase of moon? I can use those to specify the other major Holidays in India.

There's now a lunar calendar function in the `utils` module. It is based on the Chinese lunar calendar, and I am not familiar enough to know if the Indian calendar is any different.

**Comment by dr-prodigy** [comment]
Hi @rishabmehta7
sorry for getting to this so late.
At the current time, religious holidays are not taken into account, so an improvement like this, although meaningful, is out of the current scope of the lib, so it would ask for a similar review for every supported country.

Due to this, I consider it as a long running (and low-priority) task, but if you wish to work on it for India, we will then consider it / review it as a global lib improvement.

Thanks! 👍

**Comment by arkid15r** [comment]
PH supports more granular holidays definition based on region/category/special or regular status now. It should be much easier to implement those holidays support atm.

**Comment by Sheshank-singh** [comment]
I would like to work on this issue. My plan is to implement support for optional holidays specific to religions in India and address the dependency on lunar dates for certain holidays. I will explore libraries for lunar phase calculations to ensure accuracy. Please assign this to me

**Comment by arkid15r** [comment]
> I would like to work on this issue. My plan is to implement support for optional holidays specific to religions in India and address the dependency on lunar dates for certain holidays. I will explore libraries for lunar phase calculations to ensure accuracy. Please assign this to me

Thanks for your willingness to help w/ this @Sheshank-singh 
Please note this issue is pretty old and a lot has changed in the code base. We now have [Hindu calendar](https://github.com/vacanza/holidays/blob/dev/holidays/calendars/hindu.py) and [Hindu holiday group](https://github.com/vacanza/holidays/blob/dev/holidays/groups/hindu.py) that we use. The default dates can be overridden on per country basis (see example [here](https://github.com/vacanza/holidays/blob/bd18852b4d21e8c1a3559dd46bb77a3127a112c2/holidays/countries/malaysia.py#L624)).

I've assigned this to you. 

Are you going to do this as a part of Winter of Code or separately?

**Comment by Sheshank-singh** [comment]
Thank you for assigning me this task! I will follow the approach mentioned in the repository and ensure compatibility with the current codebase, particularly with the Hindu calendar and holiday group functionality. I’ll also refer to the examples for overriding default dates on a per-country basis.

To answer your question, I will work on this as a part of Winter of Code. Please let me know if there are any additional guidelines or considerations I should keep in mind

**Comment by PPsyrius** [comment]
@arkid15r @KJhellico AFAIK this has already been implemented in-full from what I understand i.e. Mahavir Jayanti in various states.

**Comment by KJhellico** [comment]
I agree. It would be good to thoroughly check India regional holidays with official sources, but that's a topic for another issue.
